python-modules/cis_profile/cis_profile/profile.py

```diff
@@ -100,6 +100,66 @@ def get_profile_from_file(self, user_structure_json_path):
             path = user_structure_json_path
         return DotDict(json.load(open(path)))
 
+    def merge(self, user_to_merge_in, publisher, level=None, _internal_level=None):
```

Comment:
I added some comments. I would carry on the `_internal level` the same way as `level`. Probably give them better names like `original_user_level` and `to_merge_in_user_level` or something.

Bigger question, do we need to merge everything from an attribute, including signatures and metadata?

I like how you deal with _null_ updated and I would add a flag to change this on demand (to enable field deletion).

Reply:
My thinking was if we plan to use this for partial updates via the change API. The updated fields would be signed by the RP (DinoPark most likely) already.
Or is this only meant to be used by an RP as a lib?